src/pymor/parameters/base.py

```diff
@@ -349,135 +350,98 @@ def __new__(cls, *args, **kwargs):
+    def __len__(self):
+        return len(self._values)
 
-        Returns
-        -------
-        If `param` depends on time, this corresponding |Function| (and not its
-        evaluation at the current time) is returned. If `param` is not given
-        by a |Function|, a |ConstantFunction| is returned.
-        """
-        from pymor.analyticalproblems.functions import Function
-        value = self._raw_values[param]
-        if not isinstance(value, Function):
-            from pymor.analyticalproblems.functions import ConstantFunction
-            value = ConstantFunction(value)
-        return value
+    def __contains__(self, key):
+        return key in self._values
+
+    def keys(self):
+        return self._values.keys()
+
+    def values(self):
+        return self._values.values()
 
-    def with_(self, **kwargs):
-        return Mu(self._raw_values, **kwargs)
+    def items(self):
+        return self._values.items()
+
+    def get(self, key, value=None):
+        return self._values.get(key, value)
 
     @property
-    def parameters(self):
-        return Parameters({k: v.size for k, v in self.items()})
+    def has_time_dependent_values(self):
+        return len(self.time_dependent_values) > 0
 
-    def allclose(self, mu):
-        """Compare dicts of |parameter values| using :meth:`~pymor.tools.floatcmp.float_cmp_all`.
+    def parameters(self, include_time_dependent=False):
+        params = {k: v.size for k, v in self.items()}
+        if include_time_dependent:
+            params.update({k: v.shape_range[0] for k, v in self.items()})
+        return Parameters(params)
 
-        Parameters
-        ----------
-        mu
-            The |parameter values| with which to compare.
+    def with_(self, new_type=None, **kwargs):
+        cls = new_type or Mu
+        return cls(**(self._values | self.time_dependent_values | kwargs))
 
-        Returns
-        -------
-        `True` if both |parameter value| dicts contain values for the same |Parameters| and all
-        components of the parameter values are almost equal, else `False`.
-        """
-        assert isinstance(mu, Mu)
-        return self.keys() == mu.keys() and all(float_cmp_all(v, mu[k]) for k, v in self.items())
+    def at_time(self, t):
+        if 't' in self:
+            raise ValueError('time already specified')
+        t = np.asarray(t).reshape((1,))
+        return Mu(self._values, t=t, **{k: v(t) for k, v in self.time_dependent_values.items()})
 
     def to_numpy(self):
-        """All parameter values as a NumPy array, ordered alphabetically."""
+        if self.has_time_dependent_values:
+            raise ValueError(f'Mu has time-dependent values ({list(self.time_dependent_values.keys())}).')
         if len(self) == 0:
             return np.array([])
         else:
-            return np.hstack([v for k, v in self.items()])
-
-    def copy(self):
-        return self
+            return np.hstack([v for k, v in self._values.items()])
```

Comment:
```suggestion
            return np.hstack([v for k, v in self.items()])
```

Reply:
This was intended as a micro optimization to avoid the additional function call. However, as it seems to be confusing, let's go with your suggestion.

Reply:
I wasn't confused, I just thought that it might look a bit cleaner. But you are right, avoiding the additional function call is definitely an argument for your version. So I would be fine with going back to the original call.